discord_slash/client.py

```diff
@@ -220,8 +221,13 @@ async def to_dict(self):
             selected = self.commands[x]
             command_dict = {
                 "name": x,
-                "description": selected.description or "No Description.",
-                "options": selected.options or []
+                "cmd": {
+                    "name": x,
+                    "description": selected.description or "No Description.",
+                    "options": selected.options or [],
+                    "default_permission": selected.default_permission
+                },
```

Comment:
I was thinking of something like this:
```py
{
  "global": # global cmds here
  "guild": {
    guildid1: {
      "cmd": # commands for that guild here
      "perms": # permissions for that guild here
    }
  }
}
```

But I think a better mapping which I am going to change to soon, is making it completely separate permissions from commands:
```py
{
  "global": # global cmds here
  "guild": {
      guildid1: # commands for that guild here
    }
  }
  "permissions: {
      guildid2: # permissions for that guild here
    }
}
```

Reply:
Flatten the dict so it's 
```py
{
  "name": name,
   "description":description,
   "options":options,
   "default_permission":default_permission,
   "permissions": permissions
}
```
I don't see any reason to nest the command data separately, you are correct that the api does not return permission data when requesting commands, but that doesn't mean that we should separate them.  
`to_dict` is a documented function and therefore changing the structure of what is returning is a breaking change.

Reply:
https://github.com/eunwoo1104/discord-py-slash-command/blob/d5cba045a60aa5f9e3e16d16e7a4520fbdd31e3c/discord_slash/client.py#L322-L336

It was due to the fact that that dict itself is supposed to be the exact thing that is sent to discord api, so I decided to split it. But its definitely possible to just do something like this in the code for loop code snippet shown above:
```
{
   "name": command.name,
   "description": command.description,
   "options": command.options,
   "default_permission": command.default_permission
}
```
Since we should not be including the permissions on the request for putting slash commands to discord api.

Reply:
imo flatten it and remove the permissions key in `sync_all_commands`. `to_dict` is meant to provide a representation of the commands in a dictionary format, not necessarily commands which are exactly in the format discord wants.
Also the api doesn't care if you include extra fields, just tested that. (still should remove `permissions` before sending the request solely to reduce data sent.)
I just think that we should avoid breaking changes :shrug: